src/components/rating/rating.tsx

```diff
@@ -0,0 +1,154 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Method,
+  Prop,
+  State
+} from "@stencil/core";
+import { IFormComponent } from "../common/interfaces";
+@Component({
+  shadow: false,
+  styleUrl: "rating.scss",
+  tag: "gx-rating"
+})
+export class Rating implements IFormComponent {
+  @State() ratingScore = 0;
+
+  @Element() element: HTMLElement;
+
+  /**
+   * This attribute allows you specify if the element is disabled.
+   * If disabled, it will not trigger any user interaction related event
+   * (for example, click event).
+   */
+  @Prop() disabled = false;
+
+  /**
+   * The control id. Must be unique per control!
+   */
+  @Prop() id: string;
+
+  /**
+   * This attribute lets you specify how this element will behave when hidden.
+   *
+   * | Value        | Details                                                                     |
+   * | ------------ | --------------------------------------------------------------------------- |
+   * | `keep-space` | The element remains in the document flow, and it does occupy space.         |
+   * | `collapse`   | The element is removed form the document flow, and it doesn't occupy space. |
+   */
+  @Prop() invisibleMode: "collapse" | "keep-space" = "collapse";
+
+  /**
+   * The current value displayed by the component.
+   *
+   */
+  @Prop() maxValue: number;
+
+  /**
+   * This attribute indicates that the user cannot modify the value of the control.
+   * Same as [readonly](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#attr-readonly)
+   * attribute for `input` elements.
+   * _Disable by default_
+   */
+  @Prop() readonly = false;
+
+  /**
+   * The current value displayed by the component.
+   *
+   */
+  @Prop() value = 0;
+
+  handleChange: (UIEvent: any) => void;
+
+  /**
+   * The 'input' event is emitted when a change to the element's value is committed by the user.
+   */
+  @Event() input: EventEmitter;
+
+  onClick(event: UIEvent) {
+    const element = event.target as HTMLElement;
+    const score =
+      element.nodeName === "polygon"
+        ? Array.from(element.parentElement.parentElement.children).indexOf(
```

Comment:
I thought that too.
The problem is that the parent of the target element is undefined. Clicking in a star the `onClick` event is triggered but the element target can be the `<polygon>` element or the `<svg>` element. The variable where the parent element will be saved will not have a specific name like: 
 ```typescript
const svgElement = element.parentElement;;
```
Doing what you correctly recommend it will be something like:
```typescript
onClick(event: UIEvent) {
    const element = event.target as HTMLElement;
    const targetParent = element.parentElement;
    element.nodeName === "polygon"
        ? Array.from(targetParent.parentElement.children).indexOf(
            targetParent
          ) + 1
        : Array.from(targetParent.children).indexOf(element) + 1;
    this.ratingScore = score;
    this.input.emit(this);
}
```